src/transformStateWithClones.js

```diff
@@ -7,7 +8,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newObject = { ...state };
+  const result = [];
+
+  actions.forEach(element => {
+    switch (element.type) {
+      case 'addProperties':
```

Comment:
Describe all cases in constant. 
For example:
const ACTION_CASES = {
  add: 'addProperties',
}